integrationtest/src/androidTest/java/com/swiftkey/cornedbeef/CoachMarkTestCase.java

```diff
@@ -107,7 +108,38 @@ public void run() {
         
         assertFalse(mCoachMark.isShowing());
     }
-    
+
+    /**
+     * Test that the popup disappears after the anchor view detaches if specified in the builder
+     */
+    public void testDismissAfterAnchorDetach() throws InterruptedException {
+        final View anchorView = new View(mActivity);
+        getInstrumentation().runOnMainSync(new Runnable() {
+            @Override
+            public void run() {
+                mActivity.setContentView(anchorView);
+            }
+        });
+
+        mCoachMark = new TestCoachMark.TestCoachMarkBuilder(
+                mActivity, anchorView, "spam spam spam").setDismissOnAnchorDetach(true).build();
+
+        showCoachMark(getInstrumentation(), mCoachMark);
+
+        assertTrue(mCoachMark.isShowing());
+
+        Thread.sleep(2000);
+
+        getInstrumentation().runOnMainSync(new Runnable() {
+            @Override
+            public void run() {
+                mActivity.setContentView(R.layout.coach_mark_test_activity);
+            }
+        });
+
+        assertFalse(mCoachMark.isShowing());
```

Comment:
What happens if setDismissOnAnchorDetach is set to false, would this be true? Or would we crash? Or something else?

Reply:
was thinking to add the other test case as well but was afraid of duplicating; will sort this out